src/main/java/us/ihmc/handsros2/abilityHand/AbilityHand.java

```diff
@@ -0,0 +1,604 @@
+ *       <li>Middle finger</li>
+ *       <li>Ring finger</li>
+ *       <li>Pinky finger</li>
+ *       <li>Thumb flexor</li>
+ *       <li>Thumb rotator</li>
+ *    </ol>
+ * </p>
+ * <p>
+ * Following the above method, the hand's fingers are specified using indices [0, 4],
+ * where 0 = index finger, 1 = middle finger, and so on for the ring, pinky, and thumb fingers.
+ * </p>
+ * <p>
+ * The hand reports actuator velocities as radians per second.
+ * </p>
+ * <p>
+ * Ability Hands can be optionally equipped with touch sensors.
+ * Force sensitive resistors (FSRs) are used to sense pressure exerted on the fingers.
+ * Each finger will have 6 sensors, making 30 total.
+ * The sensors are indexed following the order of finger indices. I.e. sensors [0, 5] are index finger sensors,
+ * [6, 11] are middle finger sensors, and so on with the thumb sensors being last.
+ * </p>
+ * <p>
+ * For more information on the Ability Hands, you can read the
+ * <a href="https://github.com/psyonicinc/ability-hand-api/blob/master/Documentation/ABILITY-HAND-ICD.pdf">Ability Hand Interface Control Document</a>.
+ * </p>
+ *
+ * <p>
+ * Additionally, this class manages higher-level control of an Ability Hand, including position, velocity,
+ * velocity-to-position, and multi-stage grip operations, with YoVariables.
+ * </p>
+ */
+public class AbilityHand implements HandInterface
+{
+   public static int ACTUATOR_COUNT = 6;
+   public static int TOUCH_SENSOR_COUNT = 30;
+
+   private static final float DEADZONE = 3.0f; // TODO: Reduce with current control
+
+   private final String identifier;
+   private final RobotSide handSide;
+
+   /** Command type used for low-level interface. */
+   private final YoEnum<AbilityHandCommandType> commandType;
+
+   /** Position command values sent to the hand. */
+   private final YoFloatArray positionCommands;
+   /** Velocity command values sent to the hand. */
+   private final YoFloatArray velocityCommands;
+   /** Filtered command values. */
+   private final YoFloatArray filteredCommandValues;
+   /** Measured actuator positions in degrees. */
+   private final YoFloatArray actuatorPositions;
+   /** Measured actuator velocities in radians per second. */
+   private final YoFloatArray actuatorVelocities;
+   /** Filtered actuator velocities in radians per second. */
+   private final YoFloatArray filteredActuatorVelocities;
+   /** Measured actuator currents in amperes. */
+   private final YoFloatArray actuatorCurrents;
+   /** Raw touch sensor FSR readings (ADC counts). */
+   private final YoIntegerArray rawFSRReadings;
+
+   /** High-level control mode (POSITION, VELOCITY, GRIP). */
+   private final YoEnum<AbilityHandControlMode> controlMode;
+   private AbilityHandControlMode previousControlMode = null;
+
+   /** High-level multi-stage grip pattern. */
+   private final YoEnum<AbilityHandGrip> grip;
+   private final YoEnum<AbilityHandGrip> previousGrip;
+   private final YoLong gripStage;
```

Comment:
This doesn't need to be a long, does it?
```suggestion
   private final YoInteger gripStage;
```